### PR TITLE
[feat] Adding typescript type definitions

### PR DESCRIPTION
**Pull request by fxdemolisher**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    model: string
+  }
+
+  export type CastState =
+    | 'NoDevicesAvailable'
+    | 'NotConnected'
+    | 'Connecting'
+    | 'Connected'
+
+  export type CastOptions = {
```
Comment (benoitdion) [inline index.d.ts]: nit: you may want to wrap this in a `Partial` instead of making each field optional

Comment (fxdemolisher) [reply index.d.ts]: Switched to `Partial` style as suggested

**Comment by fxdemolisher** [comment]
@petrbela @nbennink would it be possible to get eyes on this PR? Am i missing anything in order to get this merged in?

**Comment by nbennink** [comment]
I'm not sure if anything in `CastOptions` is required (customData for example). But looks good to me though.

**Comment by fxdemolisher** [comment]
@nbennink The `Partial` wrapper around CastOptions means every member is optional. In my opinion that makes the type a bit more readable, but I can also change that to use `?:` notation instead if that is preferred. The same approach of using `Partial` was used for `TextTrackStyle` as well.

**Comment by fxdemolisher** [comment]
@petrbela would it be possible to get a review on this PR?

File: package.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
   "version": "3.1.3",
   "description": "React Native wrapper for the Google Cast SDK for iOS and Android",
   "main": "index.js",
+  "types": "index.d.js",
```
Comment (petrbela) [inline package.json]: `index.d.ts`

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    foregroundColor: string
+    windowColor: string
+    windowCornerRadius: number
+    windowType: 'none' | 'normal' | 'rounded'
+  }>
+
+  const GoogleCast: {
+    getCastDevice(): Promise<CastDevice>
+    getCastState(): Promise<CastState>
+    castMedia(options?: CastOptions): void
```
Comment (petrbela) [inline index.d.ts]: `options` are required

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    model: string
+  }
+
+  export type CastState =
+    | 'NoDevicesAvailable'
+    | 'NotConnected'
+    | 'Connecting'
+    | 'Connected'
+
+  export type CastOptions = Partial<{
```
Comment (petrbela) [inline index.d.ts]: Please don't use `Partial`.

`mediaUrl` is required, all the others are optional.

**Comment by petrbela** [comment]
Btw v4 is written in TypeScript, so this is probably not needed, other than support for v3.

File: package.json (hunk 2 above)
Comment (fxdemolisher) [reply package.json]: doh, thanks for the catch

File: index.d.ts (hunk 4 above)
Comment (fxdemolisher) [reply index.d.ts]: corrected, is Partial still ok for TextTrackStyle? (which is not in this PR but in https://github.com/react-native-google-cast/react-native-google-cast/pull/127)

Comment (petrbela) [reply index.d.ts]: I'd use Partial (or Required/Omit/Pick) only to modify an external type.

In the case here, it's much clearer to use `?` for optional props. Also, with Partial, you can't make a field required.

Comment (fxdemolisher) [reply index.d.ts]: got it, switching away from partial. TextTrackStyle that is used in #127 has every prop optional so it is marked here that way to match.

**Pull request merged**